lib/Doctrine/MongoDB/Collection.php

```diff
@@ -146,7 +149,10 @@ public function aggregate(array $pipeline, array $options = array() /* , array $
     public function batchInsert(array &$a, array $options = array())
     {
         if ($this->eventManager->hasListeners(Events::preBatchInsert)) {
-            $this->eventManager->dispatchEvent(Events::preBatchInsert, new EventArgs($this, $a, $options));
+            $eventArgs = new EventArgs($this, $a, $options);
+            $this->eventManager->dispatchEvent(Events::preBatchInsert, $eventArgs);
+            $a = $eventArgs->getData();
```

Comment:
Since EventArgs doesn't take its arguments by reference, I wanted to confirm that this would work with `batchInsert()`'s by-reference parameter. According to https://3v4l.org/4O6VL, re-assigning `$a` seems to have the desired effect of passing the changes back on to the calling context (I'm admittedly a bit rusty on the ins and outs of PHP references :)

I'll assume this is fine for `insert()` as well.